Add answer to a recurring question re UTC (#1257)
php-fig/fig-standards: proposed/clock-meta.md

 * 
_**Note:** Order descending chronologically._

## 9. FAQ

### Why not simply convert all times to UTC?

> Timezones change! Regularily! And unforseeably! So when you are converting a DateTime in the future to UTC you are using the offset that is known today to convert from the local time to UTC. When that offset changes between now and the datetime then you have no chance to adapt to that. 

Author: That is not an issue as that information is encoded into the timezone-information.

Remember: A timezone is a geographical region that is observing the same date and time by governmental definition. Such a timezone is identified by a timezone-identifier such as `Europe/Moskow`. 
Do not confuse timezones with offsets that are defining the difference to the time-standard (which can be either GMT **or** UTC) or with named offsets like `MSK` that cover some (but not all) timezones that are observing a certain offset. A timezone can be referenced via different named offsets at different times due to DST changes.

So when using a timezone, the DST information is available. When using an offset then you do not know whether that offset is standard-time or daylight saving time.

An Example: in  summer 2016 you scheduled an event in turkey for 14:00 on the 25th of December 2016. With the information known in summer 2016 you would calculate that that is 12:00 UTC. What you couldn't imagine is, that the Government decided later that year to stay on daylight saving time and not go back to standard time for the winter. So when the time comes you would show up at 12:00 UTC. With the then relevant timezone information that would be 15:00 local time. But your appointment was at 14:00 local time. Would you have *not* converted to UTC you would have stored `14:00 Asia/Istanbul` and you would have been on time.

> **Note:** While this might only seem relevant for future datetimes it illustrates that transposiing between timezones can cause much more confusion than just keeping and using the existing timezone information. 

Author: The Term "DateTime" here refers to a defined point in Date and Time. Not an instance of `DateTime`.

> **Note:** While this might only seem relevant for future datetimes it illustrates that transposiing between timezones can cause much more confusion than just keeping and using the existing timezone information. 

Author: Unless you know off mails hitting the timezone-database mailinlist reading "We changed from Offset X to Y 2 weeks ago" or " we will change from offset x to y tomorrow" (which means a new tzdb version will not be available to machines at the time the change actually hits.)

So unless your server is actually set to observing UTC and you are 100% sure that the result of the `now` function taps directly into that and doesn't do any conversions in between using the local time might be the more accurate variant.

And regardless of that: `now()` is the equivalent of checking your clock at home or your wristwatch. Which usually shows your local date and time. And from that you can then decide for yourself what you make up with that information.

> **Note:** While this might only seem relevant for future datetimes it illustrates that transposiing between timezones can cause much more confusion than just keeping and using the existing timezone information. 

Reviewer: Maybe it's better to write "date and time" then to avoid confusion?

 * 
_**Note:** Order descending chronologically._

## 9. FAQ

### Why not simply use UTC?

There are different reasons why this interface does not enforce a specific timezone.

For one thing there is a purely technical reason. The interface provides an explicit contract. Part of this 
contract is the value that is returned by the `now`-function. The only thing that currently can explicitly 
be enforced on the language level is, that the returned value is of type `DateTimeImmutable`. There 
currently is no way to enforce that this object has a certain timezone as its value. So from a language 
level perspective there is no way to explicitly enforce a certain timezone in the returned object.

> On the other hand there is also a logical reason. The explicit contract should be usable in all situations

Reviewer: ```suggestion
On the other hand, there is also a logical reason. The explicit contract should be usable in all situations
```

where one needs a way to retrieve the current time. And on the contract level we should not make an 
assumption about what the caller might need. So would the contract define that only UTC is returned, then 
use-cases that do require something else will have to explicitly work around that or find other ways of 

> handling the problem the contract tries to solve as the contract does not fit the needs. This is different 

Reviewer: ```suggestion
handling the problem, the contract tries to solve as the contract does not fit the needs. This is different 
```

> from the issue of i.e. immutability which can also not be enforced on the language level but which is

Reviewer: ```suggestion
from the issue of i.e., immutability which can also not be enforced on the language level but which is
```

> necessary to adhere to other calls on the contract. IN the case of the `clock`-interface there will be no 

Reviewer 2: ```suggestion
necessary to adhere to other calls on the contract. In the case of the `clock`-interface there will be no 
```
Typo?

other calls.

> And most important of all, the explicit contract provided by this interface does not reduce a users

Reviewer: ```suggestion
And most important of all, the explicit contract provided by this interface does not reduce a user’s
```

possibilities to use an implicit contract within their application, that defines that the interface 
will only return `DateTimeImmutable`s with a specific timezone set. And whether that is `UTC` or reduce
`Antarctica/Troll` is completely up to the user. 

The explicit contract that is defined by the interface does not limit a user in what they are doing. It 
tries to solve the problem of getting the current time in a reliable way. Which view on the current time 
that is, is not part of the explicit contract.

> So all in all: This interface tries to be as open as possible while at the same time being as 

Reviewer: ```suggestion
So, all in all: This interface tries to be as open as possible while at the same time being as 
```
